src/Features/Core/Portable/Snippets/SnippetProviders/AbstractIfSnippetProvider.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Extensions.ContextQuery;
+using Microsoft.CodeAnalysis.Simplification;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.Snippets
+{
+    internal abstract class AbstractIfSnippetProvider : AbstractSnippetProvider
+    {
+        /// <summary>
+        /// Gets the corresponding pieces of the if statement that will be implemented differently per language.
+        /// </summary>
+        protected abstract void GetPartsOfIfStatement(SyntaxNode node, out SyntaxToken openParen, out SyntaxNode condition, out SyntaxNode statement);
+        public override string SnippetIdentifier => "if";
+
+        public override string SnippetDisplayName => FeaturesResources.Insert_an_if_statement;
```

Comment:
Do we agree with this name?

Reply:
nit: place properties above methods :)

the name seems fine with me for now.